artifactory/cli.go

```diff
@@ -1639,7 +1639,15 @@ func npmPublishCmd(c *cli.Context) error {
 		}
 		npmCmd := npm.NewNpmPublishCommand()
 		npmCmd.SetConfigFilePath(configFilePath).SetArgs(args)
-		return commands.Exec(npmCmd)
+		err = commands.Exec(npmCmd)
+		if err != nil {
+			return err
+		}
+		if npmCmd.IsDetailedSummary() {
+			result := npmCmd.Result()
+			return cliutils.PrintSummaryReport(result.SuccessCount(), result.FailCount(), result.Reader(), "", err)
+		}
+		return nil
```

Comment:
The summary should be printed in case of errors
```suggestion
		if npmCmd.IsDetailedSummary() {
			result := npmCmd.Result()
			return cliutils.PrintSummaryReport(result.SuccessCount(), result.FailCount(), result.Reader(), "", err)
		}
		return err
```